official-antistasi-community/A3-Antistasi
# Rebalance reinforcements, plus related tweaks

### What type of PR is this.
1. [X] Bug
2. [X] Change
3. [ ] Enhancement

### What have you changed and why?
- Balance reinforcement system for player count.
- Enable reinforcing from "carriers".
- Separate road patrol generation from reinforcements and rebalance.
- Sanitize garrison sizes (fewer giant and tiny garrisons, units of 4 rather than 8).
- Use more 4-man teams in garrisons for both init and reinf.
- Prevent new reinf convoys spamming after a recapture.
- Fill out the AA & AT squads with a fourth soldier, cap militia squads to 8.

The rebalanced reinforcements generate an average of around 16 troops per 10 minute tick for nine players at war level 3, or about half that for a single player. For comparison, the previous version generated 8 troops per airport (not carrier) in range. The new version is not particularly dependent on airfield count or positioning.

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Checking max garrison sizes:
`outposts apply { [_x, [_x] call A3A_fnc_garrisonSize] };`

Checking reinforcements:
`[] spawn A3A_fnc_reinforcementsAI`
Check the RPT after each call, there is logging for the total troops and for each reinforcement sent.
